### ISSUE TITLE
[RFC] upstream demo MNIST to PL

### ISSUE THREAD

**Issue by Borda** [PyTorchLightning/pytorch-lightning#12743]
## 🚀 Feature

Let's upstream a simple MNISTT demo from tests/helpers to the PL package.

### Motivation

Allow users to have a simple example as part of the distribution.
We still have `pl_examples,` but many users know about it

### Pitch

Facilitate first touch/steps with PL

### Alternatives

move it to `pl_examples`

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda

**Comment by rohitgr7** [comment]
I think we have it here: https://github.com/PyTorchLightning/pytorch-lightning/tree/master/pl_examples/basic_examples/mnist_examples

**Comment by akihironitta** [comment]
> Let's upstream a simple MNISTT demo from tests/helpers to the PL package.
> Allow users to have a simple example as part of the distribution.

Do you mean to have it inside the package and enable users to run something like this? or anything else?
```python
from pytorch_lightning import Trainer, MNISTModel

model = MNISTModel(...)
trainer = Trainer(...)
trainer.fit(model, ...)
```

**Comment by akihironitta** [comment]
This feature seems to have been addressed in #12795, which is available under `pytorch_lightning/demos/`
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/demos/boring_classes.py
